lldb/source/Plugins/SymbolFile/DWARF/DWARFIndex.h

```diff
@@ -102,6 +110,12 @@ class DWARFIndex {
   }
 
   void ReportInvalidDIERef(DIERef ref, llvm::StringRef name) const;
+
+  /// Implementation of `GetFullyQualifiedType` to check a single entry,
+  /// shareable with derived classes.
```

Comment:
I'm glad you asked, because this API has been driving me crazy 😅 
We have to return "true" if we want the "search to keep going" and false if we want the "search to stop", i.e. we found a match. This is what the callbacks are required to do as well.

If I can get support for that, I'd like to rewrite these APIs to return some kind of enumeration with two names: "KeepGoing", "Stop". Future patch though!

Reply:
> I'm glad you asked, because this API has been driving me crazy 😅 We have to return "true" if we want the "search to keep going" and false if we want the "search to stop", i.e. we found a match. This is what the callbacks are required to do as well.
> 
> If I can get support for that, I'd like to rewrite these APIs to return some kind of enumeration with two names: "KeepGoing", "Stop". Future patch though!

Yes, it would be much clearer in the code if we switched to returning an enum. Something like:
```
enum class IteratorCallbackResult {
  Continue,
  Done
}
```